contracts/math/uints.ral

```diff
@@ -343,6 +345,40 @@ Contract Uints () {
         return toU256(result)
     }
 
+    pub fn isGreaterEqual(v: U512, compareTo: U512) -> Bool {
+        if (v.higher > compareTo.higher) {
+            return true
+        }
+
+        if (v.higher == compareTo.higher && v.lower >= compareTo.lower) {
+            return true
+        }
+
+        return false
+    }
+
+    pub fn bigShl(mut v: U512, n: U256) -> U512 {
+        if (n >= WordSize * 2) {
```

Comment:
imo It's not worth it to validate this case separately, it's not practical case that will happen on the exchange